src/components/shapes/draw.js

```diff
@@ -42,7 +42,11 @@ function draw(gd) {
     // Remove previous shapes before drawing new in shapes in fullLayout.shapes
     fullLayout._shapeUpperLayer.selectAll('path').remove();
     fullLayout._shapeLowerLayer.selectAll('path').remove();
-    fullLayout._shapeSubplotLayers.selectAll('path').remove();
+
+    for(var k in fullLayout._plots) {
+        var shapelayer = fullLayout._plots[k].shapelayer;
+        if(shapelayer) shapelayer.selectAll('path').remove();
```

Comment:
Interesting - why is this better? Wasn't `fullLayout._shapeSubplotLayers` essentially just an array of all the `shapelayer` nodes? Or is the win actually below in [subroutines.js](https://github.com/plotly/plotly.js/pull/2527/commits/d3fe40dab5708bf94f83b472986726e8f8ff65a9#diff-07cd4d425af26971d4a589aaac8ea059L534) that you don't have to do a `selectAll` to find these?

Reply:
>  Or is the win actually below in subroutines.js that you don't have to do a selectAll to find these?

That. `selectAll` are bad.